test/upgrade/postdowngrade.go

```diff
@@ -48,49 +41,18 @@ func ServicePostDowngradeTest() pkgupgrade.Operation {
 
 func servicePostDowngrade(t *testing.T) {
 	t.Parallel()
-	clients := e2e.Setup(t)
-
-	names := test.ResourceNames{
-		Service: serviceName,
-	}
-
-	t.Logf("Getting service %q", names.Service)
-	svc, err := clients.ServingClient.Services.Get(context.Background(), names.Service, metav1.GetOptions{})
-	if err != nil {
-		t.Fatal("Failed to get Service:", err)
-	}
-	names.Route = serviceresourcenames.Route(svc)
-	names.Config = serviceresourcenames.Configuration(svc)
-	names.Revision = svc.Status.LatestCreatedRevisionName
-
-	url := svc.Status.URL
-
-	t.Log("Check that we can hit the old service and get the old response.")
-	assertServiceResourcesUpdated(t, clients, names, url.URL(), test.PizzaPlanetText2)
-
-	t.Log("Updating the Service to use a different image")
-	newImage := ptest.ImagePath(test.PizzaPlanet1)
-	if _, err := v1test.PatchService(t, clients, svc, rtesting.WithServiceImage(newImage)); err != nil {
-		t.Fatalf("Patch update for Service %s with new image %s failed: %v", names.Service, newImage, err)
-	}
-
-	t.Log("Since the Service was updated a new Revision will be created and the Service will be updated")
-	revisionName, err := v1test.WaitForServiceLatestRevision(clients, names)
-	if err != nil {
-		t.Fatalf("Service %s was not updated with the Revision for image %s: %v", names.Service, test.PizzaPlanet1, err)
-	}
-	names.Revision = revisionName
-
-	t.Log("When the Service reports as Ready, everything should be ready.")
-	if err := v1test.WaitForServiceState(clients.ServingClient, names.Service, v1test.IsServiceReady, "ServiceIsReady"); err != nil {
-		t.Fatalf("The Service %s was not marked as Ready to serve traffic to Revision %s: %v", names.Service, names.Revision, err)
-	}
-	assertServiceResourcesUpdated(t, clients, names, url.URL(), test.PizzaPlanetText1)
+	test.EnsureTearDown(t, e2e.Setup(t), &upgradeServiceNames)
+	updateServiceAndCheck(t,
+		upgradeServiceNames.Service,
+		test.PizzaPlanet1,
+		test.PizzaPlanetText2,
+		test.PizzaPlanetText1,
+	)
 }
 
 // CreateNewServicePostDowngradeTest verifies creating a new service after downgrade.
 func CreateNewServicePostDowngradeTest() pkgupgrade.Operation {
 	return pkgupgrade.NewOperation("CreateNewServicePostDowngradeTest", func(c pkgupgrade.Context) {
-		createNewService(postDowngradeServiceName, c.T)
+		createNewService("pizzaplanet-post-downgrade-service", c.T)
```

Comment:
why not use the const?

Reply:
Because it is used only in a single place. The const also doesn't make it clearer what this means because the name of the const was nearly the same as the string.

Reply:
On the one hand placing the const side by side with the others is nice (you see all constants in one place) but on the other hand when you're looking at this test you need to also look in the other place to find out what the string is. But I don't have a strong opinion on this.